pkg/storage/etcd3/watcher.go

```diff
@@ -180,6 +180,9 @@ func (wc *watchChan) startWatching() {
 			wc.sendEvent(parseEvent(e))
 		}
 	}
+	wc.cancel()
```

Comment:
probably we can move line 172->177 out of the for loop. Or we need to call wc.cancel right after line 176?